### PR TITLE
Added plugin OctoLight

### PR DESCRIPTION
**Pull request by gigibu5**

### CODE REVIEW

File: _plugins/octolight.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  # Compatible Python version
+  #
+  # Plugins should aim for compatibility for Python 2 and 3 for now, in which case the value should be ">=2.7,<4".
+  #
+  # Plugins that only wish to support Python 3 should set it to ">=3,<4".
+  #
+  # If your plugin only supports Python 2 (worst case, not recommended for newly developed plugins since Python 2
+  # is EOL), leave at ">=2.7,<3"
+
+  python: ">=2.7,<3"
```
Comment (jneilliii) [inline _plugins/octolight.md:80]: we no longer accept Python 2 only plugins. Your plugin needs to be updated and then this line needs to be changed to `  python: ">=2.7,<4".

File: _plugins/octolight.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+# Set this to true if your plugin uses the dependency_links setup parameter to include
+# library versions not yet published on pypi. SHOULD ONLY BE USED IF THERE IS NO OTHER OPTION!
+#follow_dependency_links: false
+
+tags:
+- raspberry pi
+- gpio
+
+screenshots:
+- url: https://github.com/gigibu5/OctoLight/blob/master/img/screenshoot.png
```
Comment (jneilliii) [inline _plugins/octolight.md:28]: Image needs to be part of PR, stored in `/assets/img/plugins/octolight/` sub folder and links updated to point there.

File: _plugins/octolight.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+#follow_dependency_links: false
+
+tags:
+- raspberry pi
+- gpio
+
+screenshots:
+- url: https://github.com/gigibu5/OctoLight/blob/master/img/screenshoot.png 
+
+featuredimage: https://github.com/gigibu5/OctoLight/blob/master/img/screenshoot.png
```
Comment (jneilliii) [inline _plugins/octolight.md:30]: Image needs to be part of PR, stored in `/assets/img/plugins/octolight/` sub folder and links updated to point there.

**Review by jneilliii — CHANGES_REQUESTED** [review]
@gigibu5 it appears that your plugin needs a couple of tweaks before we can merge. First the comment on this page, we are only accepting Python 3 compatible plugins now in the official repo. Looking at your code it seems that you have flagged it as such, so probably just needs to be fixed here in this PR.

Another change to this PR, we ask that you please include screenshots in the `/assets/img/plugins/octolight` subfolder to avoid any GPDR concerns.

We highly recommend that you have a software update check hook in your plugin, to allow the software updates to work and allow you to easily distribute updates to users that have installed the plugin and make sure they get the latest bug fixes. More information on that software update check hook can be found [here](https://docs.octoprint.org/en/master/plugins/hooks.html#octoprint-plugin-softwareupdate-check-config).

In order for your plugin to work it appears that you have a RPi.GPIO import. If that is the case, then you should include RPi.GPIO [here](https://github.com/gigibu5/OctoLight/blob/a2aea421ae088758c13495424ed8988c2c121fbb/setup.py#L36) in your plugin. So that line should be 

```
plugin_requires = ["RPi.GPIO"]
```

**Comment by gigibu5** [comment]
> We highly recommend that you have a software update check hook in your plugin, to allow the software updates to work and allow you to easily distribute updates to users that have installed the plugin and make sure they get the latest bug fixes. More information on that software update check hook can be found [here](https://docs.octoprint.org/en/master/plugins/hooks.html#octoprint-plugin-softwareupdate-check-config).

@jneilliii I have included the changes you requested. Could you please check if my update function is properly written?

**Comment by jneilliii** [comment]
@gigibu5 I don't think so. After installing the plugin, it's not listed in Software Update in the settings. Here is how I've done it, which is what's included with the plugin cookiecutter template.

https://github.com/jneilliii/OctoPrint-BLTouch/blob/dcd8475aa9c6d8af3d88d0b1c55fa78ad0092eb9/octoprint_BLTouch/__init__.py#L21-L49

**Review by jneilliii — APPROVED** [review]

**Pull request merged**